### PR TITLE
feature gate picking backends

### LINKED ISSUE #15306: Feature gate picking backends.
## What problem does this solve or what need does it fill?

A user may want to turn on picking, but disable specific picking back ends. In my case, I'd rather turn off sprite picking and use a custom, fine-grained bounding box picking backend instead. At the moment there is no way to opt-in to picking but opt-out of particular backends.

## What solution would you like?

Features seem like the right way to do this.

## What alternative(s) have you considered?

Could use a configuration resource or something, but that seems excessive. Most users probably don't need runtime configuration of picking?

### PR DESCRIPTION
**Pull request by BenjaminBrienen**
# Objective

Fixes #15306

## Solution

- Add feature gate on the module and the place where each one is used
- Declare the features and make them default

## Testing

- CI

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -87,6 +89,9 @@ default = [
   "sysinfo_plugin",
 ]
 
+bevy_sprite_picking_backend = []
```
Comment (alice-i-cecile) [inline Cargo.toml:92]: Comments please that explain the feature, like the other features.

**Review by pablo-lua — APPROVED** [review]

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Merging once you get CI passing <3

**Comment by BenjaminBrienen** [comment]
Please double check that I got the declaration of the features correct in terms of the dependencies. :)

File: crates/bevy_sprite/Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,8 @@ license = "MIT OR Apache-2.0"
 keywords = ["bevy"]
 
 [features]
-bevy_picking = ["dep:bevy_picking", "dep:bevy_window"]
+default = ["bevy_sprite_picking_backend"]
+bevy_sprite_picking_backend = ["bevy_window"]
```
Comment (alice-i-cecile) [inline crates/bevy_sprite/Cargo.toml:13]: This should also enable `bevy_picking` right?

File: crates/bevy_ui/Cargo.toml [hunk 3]
```diff
@@ -40,9 +40,9 @@ nonmax = "0.5"
 smallvec = "1.11"
 
 [features]
+default = ["bevy_ui_picking_backend"]
 serialize = ["serde", "smallvec/serde", "bevy_math/serialize"]
-bevy_picking = ["dep:bevy_picking"]
-
+bevy_ui_picking_backend = []
```
Comment (alice-i-cecile) [inline crates/bevy_ui/Cargo.toml:45]: This should enable `bevy_picking`.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Good call, I think these dependencies were misconfigured.

**Comment by BenjaminBrienen** [comment]
I don't know what dep: does. Should I add that? Seems like cargo is able to figure out whether it is a feature or a dependency.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
I think it's old syntax.

**Comment by BenjaminBrienen** [comment]
> I think it's old syntax.

Or maybe new syntax?
https://github.com/rust-lang/cargo/issues/11010
image

**Pull request merged**